sympy/physics/control/lti.py

```diff
@@ -1422,3 +1424,326 @@ def _eval_rewrite_as_TransferFunction(self, num, den, **kwargs):
 
     def __neg__(self):
         return Feedback(-self.num, self.den)
+
+
+def _to_TFM(mat, var):
+    """Private method to convert ImmutableMatrix to TransferFunctionMatrix efficiently"""
+    def group(flat, size): return [flat[i:i+size] for i in range(0, len(flat), size)]
+    tfm_cols = mat.args[1]
+    flat_list = list(mat.args[2])
+    flat_list = [TransferFunction.from_rational_expression(elem, var) for elem in flat_list]
+    arg = group(flat_list, tfm_cols)
+    return TransferFunctionMatrix(arg)
+
+
+class TransferFunctionMatrix(Basic):
+    r"""
+    A class for representing the MIMO (multiple-input and multiple-output)
+    generalization of the SISO (single-input and single-output) transfer function.
+
+    It is a matrix of transfer functions (``TransferFunction`` objects).
+    There is only one argument, ``arg`` which is also the compulsory argument.
+    ``arg`` is expected to be strictly of the type list of lists
+    which holds the transfer functions.
+
+    Parameters
+    ==========
+
+    arg: Nested ``List`` (strictly).
+        Users are expected to input a nested list of ``TransferFunction`` objects.
```

Comment:
I would have thought that a Matrix was a more natural input

Reply:
I don't get it. Creating an instance of one matrix-type class by inputting another matrix-type object seems a bit odd to me. And since non-expr objects are getting deprecated from matrix class, how can I possibly have Matrix of `TransferFunction`s as a valid input?

Reply:
If I can create a `TransferFunction` as `TransferFunction(expr, s)` then why not `TransferFunctionMatrix` as `TransferFunctionMatrix(M, s)` where `M` is a matrix of expressions involving `s`?

That seems to be how it works in Mathematica:
https://reference.wolfram.com/language/ref/TransferFunctionModel.html

Reply:
> If I can create a `TransferFunction` as `TransferFunction(expr, s)` then why not `TransferFunctionMatrix` as `TransferFunctionMatrix(M, s)` where `M` is a matrix of expressions involving `s`?
> 
> That seems to be how it works in Mathematica:
> https://reference.wolfram.com/language/ref/TransferFunctionModel.html

Oh, you meant a classmethod. I get your point now. It seems to be a good idea. @namannimmo10, what do you say? Should I make a classmethod for creating `TransferFunctionMatrix` objects this way?

Reply:
Now that we have an option to create a new transfer function from a rational expression, a similar thing can be done for TFM.